sklearn/utils/optimize.py

```diff
@@ -80,29 +139,36 @@ def newton_cg(func_grad_hess, func, grad, x0, args=(), eps=1e-4, tol=1e-4,
         ``max{|g_i | i = 1, ..., n} <= tol``
         where ``g_i`` is the i-th component of the gradient.
 
-    eps : float, optional
-        If fhess is approximated, use this value for the step size.
-
     maxiter : int
-        Number of iterations.
+        Number of Newton iterations.
+
+    maxinner : int
+        Number of CG iterations.
```

Comment:
make_classification makes a non-linear dataset so we should not increase
the maxinner because of this one. I am tempted to keep it to 100 and not
expose this param.